tests/test_xbue_calendar.py

```diff
@@ -0,0 +1,132 @@
+    HAVE_EARLY_CLOSES = True
+
+    def test_regular_holidays(self):
+        all_sessions = self.calendar.all_sessions
+
+        expected_holidays = [
+            pd.Timestamp('2019-01-01', tz=UTC),  # New Year's Day
+            pd.Timestamp('2019-03-04', tz=UTC),  # Carnival Monday
+            pd.Timestamp('2019-03-05', tz=UTC),  # Carnival Tuesday
+            pd.Timestamp('2017-03-24', tz=UTC),  # Truth and Justice Mem Day
+            pd.Timestamp('2019-04-02', tz=UTC),  # Malvinas Day
+            pd.Timestamp('2019-04-13', tz=UTC),  # Maundy Thursday
+            pd.Timestamp('2019-04-14', tz=UTC),  # Good Friday
+            pd.Timestamp('2019-05-01', tz=UTC),  # Labour Day
+            pd.Timestamp('2018-05-25', tz=UTC),  # May Day Revolution
+            pd.Timestamp('2019-06-17', tz=UTC),  # Martin Miguel de-Guemes Day
+            pd.Timestamp('2019-06-20', tz=UTC),  # National Flag Day
+            pd.Timestamp('2019-07-09', tz=UTC),  # Independence Day
+            pd.Timestamp('2019-08-19', tz=UTC),  # San Martin's Day
+            pd.Timestamp('2019-10-14', tz=UTC),  # Cultural Diversity Day
+            pd.Timestamp('2019-11-06', tz=UTC),  # Bank Holiday
+            pd.Timestamp('2019-11-18', tz=UTC),  # Day of Natl Sovereignty
+            pd.Timestamp('2017-12-08', tz=UTC),  # Immaculate Conception
+            pd.Timestamp('2019-12-25', tz=UTC),  # Christmas Day
+        ]
+
+        for holiday_label in expected_holidays:
+            self.assertNotIn(holiday_label, all_sessions)
+
+    def test_holidays_fall_on_weekend(self):
+        all_sessions = self.calendar.all_sessions
+
+        # Holidays falling on weekends are not made up, so verify surrounding
+        # days are trading days.
+        expected_sessions = [
+            # New Year's Day on Sunday, Jan 1st.
+            pd.Timestamp('2016-12-30', tz=UTC),
+            pd.Timestamp('2017-01-02', tz=UTC),
+            # Truth and Justice Memorial Day on Sunday, Mar 24th.
+            pd.Timestamp('2019-03-22', tz=UTC),
+            pd.Timestamp('2019-03-25', tz=UTC),
+            # Malvinas Day on Saturday, Apr 2nd.
+            pd.Timestamp('2016-04-01', tz=UTC),
+            pd.Timestamp('2016-04-04', tz=UTC),
+            # Labour Day on Sunday, May 1st.
+            pd.Timestamp('2016-04-29', tz=UTC),
+            pd.Timestamp('2016-05-02', tz=UTC),
+            # May Day Revolution on Saturday, May 25th.
+            pd.Timestamp('2019-05-24', tz=UTC),
+            pd.Timestamp('2019-05-27', tz=UTC),
+            # Martin Miguel de-Guemes Day on Sunday, Jun 17th.
+            pd.Timestamp('2018-06-15', tz=UTC),
+            pd.Timestamp('2018-06-18', tz=UTC),
+            # National Flag Day on Saturday, Jun 20th.
+            pd.Timestamp('2015-06-19', tz=UTC),
+            pd.Timestamp('2015-06-22', tz=UTC),
+            # Independence Day on Saturday, Jul 9th.
+            pd.Timestamp('2016-07-07', tz=UTC),
```

Comment:
Did you mean `pd.Timestamp('2016-07-08', tz=UTC)` here?

Reply:
I had bumped it back a day because there was an adhoc on the 8th, but I decided to just change it to 2017 to avoid that issue.